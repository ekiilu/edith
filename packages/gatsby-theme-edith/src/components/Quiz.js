import React from "react"

import Question from "./Question"

export default props => {
  const quiz = JSON.parse(props.children)

  return (
    <div
      style={{
        marginTop: 70,
        marginBottom: 70,
      }}
    >
      {quiz.questions.map((question, index) => {
        return <Question key={index} question={question} />
      })}
    </div>
  )
}
