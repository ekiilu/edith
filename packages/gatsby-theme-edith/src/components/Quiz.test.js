import React from "react"
import { render } from "@testing-library/react"

import Quiz from "./Quiz"

describe("Quiz", () => {
  it("renders correctly", () => {
    const question = {
      question: "Foo?",
      answers: [
        {
          answer: "Bar!",
          correct: true,
          response: "Right",
        },
        {
          answer: "Baz!",
          response: "Wrong",
        },
      ],
    }
    const json = `
    {
        "questions": [
            {
                "question": "Foo?",
                "answers": [
                    {
                        "answer": "Bar!",
                        "correct": true,
                        "response": "Right"
                    },
                    {
                        "answer": "Baz!",
                        "response": "Wrong"
                    }
                ]
            }
        ]
    }
    `
    const { asFragment } = render(<Quiz>{json}</Quiz>)
    expect(asFragment()).toMatchSnapshot()
  })
})
