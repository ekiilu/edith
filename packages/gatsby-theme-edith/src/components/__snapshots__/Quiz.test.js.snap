// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz renders correctly 1`] = `
<DocumentFragment>
  <div
    style="margin-top: 70px; margin-bottom: 70px;"
  >
    <div
      style="margin-top: 28px; margin-bottom: 28px;"
    >
      <p
        class="MuiTypography-root makeStyles-label-2 MuiTypography-body1 MuiTypography-paragraph"
      >
        Foo?
      </p>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        style="margin-bottom: 14px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-root-137 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Bar!
                </p>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-root-137 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Baz!
                </p>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
